gadget/install/install.go

```diff
@@ -307,11 +326,20 @@ func createEncryptionParams(encTyp secboot.EncryptionType) gadget.StructureEncry
 	return gadget.StructureEncryptionParameters{}
 }
 
+func onDiskStructsSortedIdx(vss map[int]*gadget.OnDiskStructure) []int {
```

Comment:
this could return perhaps the sorted structures instead of just the indexes, or use a channel to do that as an iterator